presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -190,6 +190,29 @@ public void testFiltersAndProjections()
         assertQuery("SELECT if(orderkey % 2 = 0, quantity_by_linenumber) FROM orders_ex");
     }
 
+    @Test
+    public void testAnalyzeStats()
+    {
+        assertQuery("ANALYZE region");
+        assertQuery("SHOW STATS FOR region");
+        // Create a partitioned table and run analyze on it.
+        String tmpTableName = generateRandomTableName();
+        try {
+            getQueryRunner().execute(getSession(), String.format("CREATE TABLE %s (name VARCHAR, regionkey BIGINT," +
+                    "nationkey BIGINT) WITH (partitioned_by = ARRAY['regionkey','nationkey'])", tmpTableName));
+            Session writeSession = buildSessionForTableWrite();
+            getQueryRunner().execute(writeSession,
+                    String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName));
```

Comment:
"nation" is a very tiny table for this kind of test. But its probably sufficient to test the correctness of this work.